src/systems/leader1_proximity_attack.rs

```diff
@@ -15,37 +15,43 @@ pub fn leader1_proximity_attack_system(
 ) -> SystemResult {
     let mut v = vec![];
     let mut rng = thread_rng();
-    for (e, _proximity, stat, pos, team) in
+    for (e, proximity, stat, pos, team) in
         join!(&entities && &proximity_attacks && &stats && &positions && &teams)
     {
-        let closest = find_closest_in_other_team(
-            team.unwrap(),
-            pos.unwrap(),
-            &teams,
-            &positions,
-            &stats,
-            &entities,
-        );
-        if let Some((target, _)) = closest {
-            let damage = stat.unwrap().stats.get(&Stats::Attack).unwrap().value;
-            v.push((e.unwrap().clone(), target.clone(), damage));
+        if let ProximityAttackSystems::Leader1ProximityAttack(radius) = proximity.unwrap() {
```

Comment:
The fact that the component/enum contains the word Systems is quite confusing. Could we change it to something else?